src/txdb.cpp

```diff
@@ -72,6 +96,33 @@ uint256 CCoinsViewDB::GetBestBlock() const
     return hashBestChain;
 }
 
+void CCoinsViewDB::WriteBestBlockSeq(const uint256 &hashBlock)
+{
+    LOCK(cs_utxo);
+    if (!hashBlock.IsNull())
+    {
+        db.Write(DB_BEST_BLOCK, hashBlock);
+    }
+}
+
+uint256 CCoinsViewDB::GetBestBlockDb() const
+{
+    LOCK(cs_utxo);
+    uint256 hashBestChain;
+    if (!db.Read(DB_BEST_BLOCK_BLOCKDB, hashBestChain))
```

Comment:
If the `db.Read(..)` fails, will that touch `hashBestChain` at all? If not, this code seems, except for the locking, quite cumbersome, doesn't it? (Sorry, I am by far not a `leveldb` expert...)

Reply:
the code you are questioning here is much older than this pr, im unsure what the original reasons for this were